frontend/src/app/modules/external-booking/external-booking-choose-time-page/external-booking-choose-time-page.component.ts

```diff
@@ -164,22 +174,38 @@ export class ExternalBookingTimeComponent extends BaseComponent implements OnIni
         );
     }
 
-    public isDateInRange(date: Date, min: Date, max: Date, daysToAdd: number = 0, timesToAdd = 0): boolean {
+    public convertDate(date: Date, daysToAdd: number = 0, timesToAdd = 0) {
         const firstCalendarDay = new Date(this.calendarWeek.firstDay);
 
         firstCalendarDay.setHours(date.getHours(), date.getMinutes(), date.getSeconds(), date.getMilliseconds());
         const result = addDays(firstCalendarDay, daysToAdd);
 
         result.setTime(result.getTime() + this.selectedMeetingDuration * timesToAdd * 60 * 1000);
 
-        if (this.disabledDays.includes(result.getDay())) {
-            return false;
-        }
+        return result;
+    }
 
+    public checkBookedDates(result: Date) {
         const checkBookedDate = (parsedDate: Date) =>
             parsedDate.getDate() === result.getDate() && parsedDate.getTime() === result.getTime();
 
-        if (this.orderedTimes.some((x) => checkBookedDate(new Date(Date.parse(x.startTime))))) {
+        return this.orderedTimes.some((x) => checkBookedDate(new Date(Date.parse(x.startTime))));
+    }
+
+    public isOrderedDate(date: Date, daysToAdd: number = 0, timesToAdd = 0) {
+        const result = this.convertDate(date, daysToAdd, timesToAdd);
+
+        return this.checkBookedDates(result) && result.getDate() >= this.nowDate.getDate();
+    }
+
+    public isDateInRange(date: Date, min: Date, max: Date, daysToAdd: number = 0, timesToAdd = 0): boolean {
+        const result = this.convertDate(date, daysToAdd, timesToAdd);
+
+        if (this.disabledDays.includes(result.getDay())) {
```

Comment:
this condition can be combined with if below: if (this.disabledDays.includes(result.getDay()) || this.checkBookedDates(result))